packages/toolpad-app/src/runtime/PreviewHeader.tsx

```diff
@@ -1,10 +1,178 @@
 import * as React from 'react';
-import { Button, Typography, Box, useTheme, Alert } from '@mui/material';
+import {
+  Button,
+  Typography,
+  Box,
+  useTheme,
+  Alert,
+  ButtonProps,
+  Popover,
+  styled,
+  IconButton,
+  Tooltip,
+  Snackbar,
+  IconButtonProps,
+  popoverClasses,
+} from '@mui/material';
 import EditIcon from '@mui/icons-material/Edit';
 import { useMatch } from 'react-router-dom';
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
 import { IS_CUSTOM_SERVER, PREVIEW_HEADER_HEIGHT } from './constants';
 
-export default function PreviewHeader() {
+interface CopyToClipboardButtonProps extends IconButtonProps {
+  content: string;
+}
+
+function CopyToClipboardButton({ content, onClick, ...props }: CopyToClipboardButtonProps) {
```

Comment:
Looks like this could be a useful shared component besides being just in this file?

Reply:
I don't want to incentivize maintainers importing from outside the runtime folder. That's why I opted for this setup. Maybe in the future.

Reply:
Ah got it because it's in the runtime, ok we can keep it.